api/envoy/config/filter/http/transformation/v2/transformation_filter.proto

```diff
@@ -162,15 +175,31 @@ message Extraction {
     google.protobuf.Empty body = 4;
   }
 
-  // Only strings matching this regular expression will be part of the
-  // extraction. The most simple value for this field is '.*', which matches the
-  // whole source. The field is required. If extraction fails the result is an
-  // empty value.
+  // The regex field specifies the regular expression used for matching against the source content.
+  // - In EXTRACT mode, the entire source must match the regex. The subgroup-th capturing group,
+  //   if specified, determines which part of the match is extracted.
+  // - In SINGLE_REPLACE mode, the regex also needs to match the entire source. The subgroup-th capturing group
+  //   is targeted for replacement with the replacement_text.
+  // - In REPLACE_ALL mode, the regex is applied repeatedly to find all occurrences within the source that match.
+  //   Each matching occurrence is replaced with the replacement_text, and the subgroup field is not used.
+  // This field is required, and if the regex does not match the source as per the selected mode, the result of
+  // the extraction will be an empty value.
   string regex = 2;
 
   // If your regex contains capturing groups, use this field to determine which
   // group should be selected.
+  // For EXTRACT and SINGLE_REPLACE, refers to the portion of the text
+  // to extract/replace.
+  // Config will be rejected if this is specified in REPLACE_ALL mode.
   uint32 subgroup = 3;
+
+  // The string to replace the matched portion of the source with.
+  // Used in SINGLE_REPLACE and REPLACE_ALL modes.
+  google.protobuf.StringValue replacement_text = 5;
```

Comment:
this value goes directly into regex_replace. It has semantics that we should document. see the example here: https://en.cppreference.com/w/cpp/regex/regex_replace

Reply:
Updated this here: a4ed34d857bbefa91f2984df2ff57439cb79280a 
```proto
  // Used in SINGLE_REPLACE and REPLACE_ALL modes.
  // `replacement_text` is used to format the substitution for matched sequences in the input string
  // - In SINGLE_REPLACE mode, the content in the subgroup-th capturing group is replaced with the `replacement_text`.
  // - In REPLACE_ALL mode, each sequence matching the specified regex in the in the input is replaced with the `replacement_text`.
  //     The replacement_text may contain special syntax, such as $1, $2, etc., to refer to captured groups within the regular expression.
  //     The value contained within `replacement_text` is treated as a string, and is passed to std::regex_replace as the replacement string.
  //     see https://en.cppreference.com/w/cpp/regex/regex_replace for more details.
```

I also added a test demonstrating some of this functionality here: ef5b4d9f594199f07b4ac26141f1ff4b84d36232